scorecard/profile_data.py

```diff
@@ -22,58 +21,361 @@
 
 
 class MuniApiClient(object):
-    def __init__(self, geo_code):
+    def __init__(self):
         self.API_URL = settings.API_URL + "/cubes/"
-        self.geo_code = str(geo_code)
-        self.years = YEARS
-        self.current_year = YEARS[:1]
-
-        self.queries = self.get_queries()
-        self.results = defaultdict(dict)
-
-        responses = []
         self.session = FuturesSession(executor=EXECUTOR)
-        for query_name, query in self.queries.iteritems():
-            responses.append((query_name, query, self.api_get(query)))
-
-        for (query_name, query, response) in responses:
-            self.results[query_name] = \
-                self.response_to_results(response, query)
 
     def api_get(self, query):
         if query['query_type'] == 'aggregate':
             url = self.API_URL + query['cube'] + '/aggregate'
             params = {
                 'aggregates': query['aggregate'],
-                'cut': format_cut_param(query['cut']),
+                'cut': self.format_cut_param(query['cut']),
                 'drilldown': '|'.join(query['drilldown']),
                 'order': 'financial_year_end.year:desc',
                 'page': 0,
             }
         elif query['query_type'] == 'facts':
             url = self.API_URL + query['cube'] + '/facts'
             params = {
-                'cut': format_cut_param(query['cut']),
+                'cut': self.format_cut_param(query['cut']),
                 'fields': ','.join(field for field in query['fields']),
                 'page': 0
             }
         elif query['query_type'] == 'model':
             url = self.API_URL + query['cube'] + '/model'
             params = {}
-        return self.session.get(url, params=params, verify=False)
 
-    def response_to_results(self, api_response, query):
-        api_response.result().raise_for_status()
-        response_dict = api_response.result().json()
+        response = self.session.get(url, params=params, verify=False)
```

Comment:
I suspect calling get() and result() in the same loop will lose the benefit of futures. That's why we do one loop of get()s and another loop of result()s